full-service/src/service/transaction.rs

```diff
@@ -1674,4 +1711,338 @@ mod tests {
+        let balance = service
+            .get_balance_for_address(&bob_subaddress.public_address_b58)
+            .unwrap();
+        let balance_pmob = balance.get(&Mob::ID).unwrap();
+        assert_eq!(balance_pmob.unspent, (242 * MOB) as u128);
+
+        // Attempt to spend more than Alice has (but enough that the wallet has) to Bob
+        // and trigger InsufficientFunds Error
+        let res = service
+            .build_sign_and_submit_transaction(
+                &exchange_account.id,
+                &[(
+                    bob_subaddress.public_address_b58.clone(),
+                    AmountJSON::new(250 * MOB, Mob::ID),
+                )],
+                None,
+                None,
+                None,
+                None,
+                None,
+                None,
+                TransactionMemo::RTH {
+                    subaddress_index: Some(alice_subaddress.subaddress_index as u64),
+                },
+                None,
+                Some(alice_subaddress.public_address_b58.clone()),
+            )
+            .await;
+        match res {
+            Err(TransactionServiceError::TransactionBuilder(
+                WalletTransactionBuilderError::WalletDb(
+                    WalletDbError::InsufficientFundsUnderMaxSpendable(_),
+                ),
+            )) => {}
+            Ok(_) => panic!("Should error with InsufficientFundsUnderMaxSpendable"),
+            Err(e) => panic!(
+                "Should error with InsufficientFundsUnderMaxSpendable but got {:?}",
+                e
+            ),
+        }
+
+        // Attempt to spend more than Alice has (but enough that the wallet has) to Bob
+        // and trigger InsufficientFunds Error
+        let res = service
+            .build_sign_and_submit_transaction(
+                &exchange_account.id,
+                &[(
+                    bob_subaddress.public_address_b58.clone(),
+                    AmountJSON::new(66 * MOB, Mob::ID),
+                )],
+                None,
+                None,
+                None,
+                None,
+                None,
+                None,
+                TransactionMemo::RTH {
+                    subaddress_index: Some(alice_subaddress.subaddress_index as u64),
+                },
+                None,
+                Some(alice_subaddress.public_address_b58.clone()),
+            )
+            .await;
+        match res {
+            Err(TransactionServiceError::TransactionBuilder(
+                WalletTransactionBuilderError::WalletDb(
+                    WalletDbError::InsufficientFundsUnderMaxSpendable(_),
+                ),
+            )) => {}
+            Ok(_) => panic!("Should error with InsufficientFundsUnderMaxSpendable"),
+            Err(e) => panic!(
+                "Should error with InsufficientFundsUnderMaxSpendable but got {:?}",
+                e
+            ),
+        }
```

Comment:
How is this different than the failed sign and submit above?
should this go away?
if really desired to have multiple values perhaps an iterator of the values
```suggestion
```

Reply:
Ah, I checked the narrative at the top - the goals for the two tests are:

1. Confirm that more than Alice OR Bob has, but enough that THE WALLET has (Alice + Bob + Carol)
2. More than Alice has, but enough that BOB has

Will tidy up!

Reply:
Trying to iteratorize this, and not sure how to handle the await, if you have a sug, @nick-mobilecoin !

Reply:
Ok, I think I got it with `join_all` - commit coming shortly

Reply:
Actually I don't got it - pushed my changes for you to review, if you have a sug on how to fix it up!